cell_based/src/writers/population_writers/VertexT1SwapLocationsWriter.cpp

```diff
@@ -84,6 +85,11 @@ void VertexT1SwapLocationsWriter<ELEMENT_DIM, SPACE_DIM>::Visit(VertexBasedCellP
     pCellPopulation->rGetMesh().GetOperationRecorder()->ClearT1SwapsInfo();
 }
 
+template<unsigned ELEMENT_DIM, unsigned SPACE_DIM>
```

Comment:
Surprised to see all this stuff about Vertex model T1 swaps and things needing to know anything about ImmersedBoundary stuff, related to the comment above it is suggesting something a bit unoptimal in the code organisation if it has to be here.